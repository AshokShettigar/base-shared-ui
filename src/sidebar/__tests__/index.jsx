import { shallow } from 'enzyme';
import { Sidebar } from 'platform-common-ui';

describe('sidebar', function() {
    it('should render the sidebar with open state', function() {
        const wrapper = shallow(
            <Sidebar onToggleSidebar={() => true} isOpen>
                <div>content</div>
            </Sidebar>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the sidebar with closed state', function() {
        const wrapper = shallow(
            <Sidebar onToggleSidebar={() => true} isOpen={false}>
                <div>content</div>
            </Sidebar>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should call toggleSidebarHandler function when Icon is clicked', function() {
        const toggleSidebarHandler = jest.fn();
        const wrapper = shallow(
            <Sidebar onToggleSidebar={toggleSidebarHandler} isOpen>
                <div>content</div>
            </Sidebar>
        );
        wrapper.find('.cn-sidebar__close-btn').simulate('click');
        expect(toggleSidebarHandler).toHaveBeenCalledTimes(1);
    });

    it('should call toggleSidebarHandler function when Icon is clicked', function() {
        const toggleSidebarHandler = jest.fn();
        const wrapper = shallow(
            <Sidebar onToggleSidebar={toggleSidebarHandler} isOpen={false}>
                <div>content</div>
            </Sidebar>
        );
        wrapper.find('.cn-sidebar__close-btn').simulate('click');
        expect(toggleSidebarHandler).toHaveBeenCalledTimes(1);
    });
});
