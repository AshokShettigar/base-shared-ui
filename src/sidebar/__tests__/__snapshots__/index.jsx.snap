// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar should render the sidebar with closed state 1`] = `
<div
  className="cn-sidebar cn-sidebar--closed"
  data-component="Sidebar"
>
  <div
    className="cn-sidebar__controls"
  >
    <Icon
      className="cn-sidebar__close-btn"
      icon="carat-right"
      onClick={[Function]}
      size="extra-small"
    />
  </div>
</div>
`;

exports[`sidebar should render the sidebar with open state 1`] = `
<div
  className="cn-sidebar"
  data-component="Sidebar"
>
  <div
    className="cn-sidebar__controls"
  >
    <Icon
      className="cn-sidebar__close-btn"
      icon="close"
      onClick={[Function]}
      size="extra-small"
    />
  </div>
  <div
    className="cn-sidebar__content"
  >
    <div>
      content
    </div>
  </div>
</div>
`;
